### PR TITLE
Week8, JWT Basics

### PR DESCRIPTION
**Pull request by SergeDot**

### CODE REVIEW

File: 05-JWT-Basics/preferred/public/files/tp.svg
Comment (ethanarns) [inline 05-JWT-Basics/preferred/public/files/tp.svg:1]: This theme as a whole was a really clever idea. I love it!

File: 05-JWT-Basics/preferred/controllers/main.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import jwt from 'jsonwebtoken';
+import { BadRequestError } from '../errors/bad-request.js';
+
+const logon = async (req, res) => {
+  const { name } = req.body;
```
Comment (ethanarns) [inline 05-JWT-Basics/preferred/controllers/main.js]: You'll need both name and password for this assignment. You got `name` pulling correctly though!

File: 05-JWT-Basics/preferred/controllers/main.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  if (!name) {
+    throw new BadRequestError(`Oh, please give me your name!`, 400);
+  };
+
+  const token = jwt.sign({ name }, process.env.JWT_SECRET, { expiresIn: process.env.TOKEN_EXPIRATION });
```
Comment (ethanarns) [inline 05-JWT-Basics/preferred/controllers/main.js]: This is smart, good security practice

**Review by ethanarns — CHANGES_REQUESTED** [review]
Well done overall, but a password is also very important to this. Everything else seems to be good though, should be a quick turnaround.

File: 05-JWT-Basics/preferred/controllers/main.js (hunk 1 above)
Comment (SergeDot) [reply 05-JWT-Basics/preferred/controllers/main.js]: Done, added the password functionality

**Pull request merged**